x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/InvalidMappedField.java

```diff
@@ -130,7 +130,13 @@ private static String makeErrorMessage(Map<String, Set<String>> typesToIndices)
             errorMessage.append("[");
             errorMessage.append(e.getKey());
             errorMessage.append("] in ");
-            errorMessage.append(e.getValue());
+            if (e.getValue().size() <= 3) {
+                errorMessage.append(e.getValue());
+            } else {
+                errorMessage.append(e.getValue().stream().sorted().limit(3).collect(Collectors.toList()));
```

Comment:
Since we're worried about very long lists of indexes, perhaps streaming and sorting these is not ideal? Removing the `sort()` should probably keep the stream shallow.

Reply:
Although I realize this is harder to test, due to lack of deterministic ordering in sets.

Reply:
I like the first three bit to be honest. The raw number of indices here isn't the problem so much as the length of the names times the raw number. But I think it's fine to sort them and grab the first three.